src/pwm.rs

```diff
@@ -209,6 +234,93 @@ pub struct C2;
 pub struct C3;
 pub struct C4;
 
+/*
+  The following implemention of the embedded_hal::Pwm uses Hertz as a time type.  This was choosen 
+  because of the timescales of operationsbeing on the order of nanoseconds and not being able to 
+  efficently represent a float on the hardware.  It might be possible to change the time type to
+  a different time based using such as the nanosecond.  The issue with doing so is that the max 
+  delay would then be at just a little over 2 seconds because of the 32 bit depth of the number.
+  Using milliseconds is also an option, however, using this as a base unit means that only there 
+  could be resolution issues when trying to get a specific value, because of the integer nature.
+
+  To find a middle ground, the Hertz type is used as a base here and the Into trait has been 
+  defined for several base time units.  This will allow for calling the set_period method with 
+  something that is natural to both the MCU and the end user.
+*/
+macro_rules! pwm_impl {
+    ( $( $TIMX:ident, $timX:ident, ( $($CHNUM:tt),+ ), ( $($ENCHX:ident),+ ); )+ )  => {
+            $(
+            #[allow(unused_parens)]
```

Comment:
I'm not a fan of this, where is it needed?

Reply:
Are you meaning the `#[allow(unused_parens)]`?  It might not need to be here.  This came from the macro defined at the top of the file.  I agree, if it does not need to be here, lets get rid of it.  I will look at it tonight.

Reply:
Removed